src/lava/networks/gradedvecnetwork.py

```diff
@@ -0,0 +1,273 @@
+# Copyright (C) 2022-23 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+# See: https://spdx.org/licenses/
+
+import numpy as np
+from scipy.sparse import csr_matrix
+
+from lava.proc.graded.process import InvSqrt
+from lava.proc.graded.process import NormVecDelay
+from lava.proc.sparse.process import Sparse
+from lava.proc.dense.process import Dense
+from lava.proc.prodneuron.process import ProdNeuron
+from lava.proc.graded.process import GradedVec
+from lava.proc.lif.process import LIF
+from lava.proc.io import sink, source
+from lava.proc import embedded_io as eio
+
+from lava.magma.core.process.variable import Var
+from lava.magma.core.process.ports.ports import InPort, OutPort
+
+from .network import AlgebraicVector, AlgebraicMatrix
+
+
+class InputVec(AlgebraicVector):
+    """InputVec
+    Simple input vector. Adds algebraic syntax to RingBuffer
+
+    Parameters
+    ----------
+    vec: np.ndarray
+        NxM array of input values. Input will repeat every M steps.
+    loihi2=False: bool
+        Flag to create the adapters for loihi 2.
+    """
+
+    def __init__(self, vec, **kwargs):
```

Comment:
Please add any parameter used in the `__init__` method to the signature.
For instance, the `loihi2` and `exp` parameters are not present.

Also, don't forget type hints!